components/script/dom/xmlhttprequest.rs

```diff
@@ -255,6 +261,18 @@ impl XMLHttpRequest {
                 let rv = self.xhr.root().process_response_complete(self.gen_id, response);
                 *self.sync_status.borrow_mut() = Some(rv);
             }
+
+            fn resource_timing(&mut self) -> &mut ResourceFetchTiming {
+                &mut self.resource_timing
+            }
+
+            fn submit_resource_timing(&self) {
+                let local_name = DOMString::from("xmlhttprequest");
+                let global = self.global.root();
+                let performance_entry = PerformanceResourceTiming::new(
+                    &global, global.get_url().clone(), local_name, None, &self.resource_timing);
```

Comment:
We should store the url from RequestInit in XHRContext instead of using the global's URL here.